feat: Authentication System Implementation with OpenIddict Integration

bewitchedohana opened the pull request:
## Overview
This PR introduces a comprehensive authentication system implementation using OpenIddict, built with clean architecture principles. The system provides a foundation for user management and authentication workflows in the StarKeeper application.

## Key Implementation Details
- Created a multi-layer architecture (Application, Domain, Infrastructure, Persistence, Presentation)
- Implemented user management features with validation
- Set up authentication and authorization controllers
- Integrated Entity Framework with initial database schema
- Developed web UI components for user authentication flows
- Configured OpenIddict boilerplate for OAuth/OpenID Connect capabilities

## Technical Details
- Uses Clean Architecture patterns for better separation of concerns
- Implements command pattern with validation pipeline
- Includes unit tests for core user management functionality
- Features a Bootstrap-based UI for login and registration
- Set up initial database migration for user management

## Testing
The implementation includes unit tests covering the user creation workflow. All tests are passing.

## Future Considerations
This implementation sets the groundwork for:
- OAuth2/OpenID Connect flows
- Additional authentication features
- Extended user management capabilities

## Additional Notes
Backend solution structure has been set up with proper gitignore configuration for .NET development.

notshowporcupinefrog commented on backend/src/IdentityService/IdentityService.Infrastructure/Workers/ClientWorker.cs:
```diff
@@ -0,0 +1,54 @@
+
+    public async Task StartAsync(CancellationToken cancellationToken)
+    {
+        // TODO: Improve this to match actual initialization logic
```
Ideally, these TODOs would have an issue related, so we would be able to track them with ease

notshowporcupinefrog commented on backend/src/IdentityService/IdentityService.Presentation/Controllers/RegistrationController.cs:
```diff
@@ -0,0 +1,42 @@
+                return RedirectToAction(nameof(Index));
+            }
+
+            // TODO: Redirect to user page
```
Same case as before

notshowporcupinefrog commented on backend/src/IdentityService/IdentityService.Presentation/Program.cs:
```diff
@@ -0,0 +1,36 @@
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
```
These unnecessary comments should be removed

notshowporcupinefrog commented on backend/src/IdentityService/IdentityService.Presentation/Views/Authorization/Authorize.cshtml.cs:
If this file is not needed, it should be removed

notshowporcupinefrog commented on backend/tests/IdentityServiceTests/IdentityService.Application.Tests/Unit/CreateUserCommandHandlerTests.cs:
This file is full of unnecessary line breaks (three instead of one)

notshowporcupinefrog commented on compose.dev.yaml:
```diff
@@ -0,0 +1,18 @@
+services:
+  identity-database:
+    image: postgres:15
```
Even though these secrets are for development, they would be better on an env file, instead of hardcoded.
We should open an issue to dockerize the app and this should be on it.

bewitchedohana committed:
chore: added gitignore

Adds the gitignore to the project, so it won't have unwanted files from .NET

bewitchedohana committed:
chore: created backend solution

bewitchedohana committed:
feat: initial Identity Service implementation

Establish the Identity Service foundation using clean architecture principles,
implementing a complete user management system. The service includes user creation
functionality with validation, authentication, and authorization capabilities
through dedicated controllers. Set up Entity Framework infrastructure with initial
database migration and configured a web-based user interface featuring
registration pages styled with Bootstrap. Testing infrastructure is in place with
initial unit tests for user creation workflow. It also sets up the boilerplate for
the OpenIddict, so it's easier to work on it

notshowporcupinefrog reviewed (approved):
LGTM
